src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
```

Comment:
I added to this component a useRef to avoid the initial render triggering a `dispatch()` of an updatePlotConfig action.

I believe this is not really a big problem, but it does look a little bit weird now compared to before, because now each render of the heatmap plot is very easy to notice (due to the random sampling of cells from each cell set), so the plot would change visibly when these settings were opened initially.

I also did this in the groupBy settings, which had the same issue.

Reply:
Good spot about unneeded triggering of `dispatch`, however this doesn't seem like the right way to avoid it. The two `useEffect` functions within which you have placed the `isInitialRenderRef.current` check to avoid executing, get triggered because whatever is in the square (`[]`) brackets for the respective useEffect has been changed, which in this case is `cellSets.hierarchy` and `trackData`. In order to prevent the trigger of the dispatch for `updatePlotConfig`, you should look at the relevant `useEffect` function and prevent `trackData` from changing on initial render. In that way, the dispatch will not happen.

Reply:
I would recommend that you read a bit more about `useEffect` and initial component rendering, it will be very helpful for this task.

Reply:
Hmm, actually you are right, I had a read around that people tend to use useRef for this: https://stackoverflow.com/questions/53253940/make-react-useeffect-hook-not-run-on-initial-render, https://medium.com/swlh/prevent-useeffects-callback-firing-during-initial-render-the-armchair-critic-f71bc0e03536

In terms of the weirdness that you mention that is caused by random sampling, double check with Vicky if it is a problem.